servers/plugins/org.fusesource.ide.server.karaf.core/src/org/fusesource/ide/server/karaf/core/server/subsystems/Karaf2xPublishController.java

```diff
@@ -156,4 +191,39 @@ public void publishServer(int kind, IProgressMonitor monitor)
 			throws CoreException {
 		validate();
 	}
+	
+	/*
+	 * The following two methods are used when delegating publishing to an existing publish controller,
+	 * while still performing the final operation of transfer or remove. 
+	 * 
+	 * Basically, it allows for a delegate publish controller to build a module (such as an osgi project)
+	 * even when it doesn't actually know how to transfer it to the server. 
+	 */
+
+	
+
+	@Override
+	public int transferBuiltModule(IModule[] module, IPath srcFile, IProgressMonitor monitor) throws CoreException {
+		BundleDetails bd = new ModuleBundleVersionUtility().getBundleDetails(module, srcFile);
+		int status = IServer.PUBLISH_FULL;
+		if( bd != null ) {
+			status = getPublisher(module).publish(getServer(), module, bd.getSymbolicName(), bd.getVersion(), srcFile);
+			((Server)getServer()).setModuleState(module, status);
+			((Server)getServer()).setModulePublishState(module, IServer.PUBLISH_STATE_NONE);
+			((Server)getServer()).setServerPublishState(IServer.PUBLISH_STATE_NONE);			
+			status = IServer.PUBLISH_STATE_NONE;
+		}
+		return status;
+	}
+
+	@Override
+	public int removeModule(IModule[] module, IProgressMonitor monitor) throws CoreException {
+		BundleDetails bd = new ModuleBundleVersionUtility().getBundleDetails(module, null);  // This will clearly break... wtf.  Crap. 
```

Comment:
It will clearly break bc passing null into the second parameter will make it impossible to find the osgi version details needed.